### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by tmat**
Add links to NuGet feeds, reorder sections - move CI status to the bottom.

### CODE REVIEW

**Comment by tmat** [comment]
@jaredpar @jinujoseph @mikadumont @kendrahavens @JoeRobich @jasonmalinowski PTAL

File: README.md [hunk 1]
```diff
@@ -53,6 +23,10 @@ To install the latest release without Visual Studio, download the [.NET SDK nigh
 
 See [what's new with the C# and VB compilers](https://github.com/dotnet/roslyn/wiki/Changelog-for-C%23-and-VB-compilers).
 
+**Pre-release builds** are available from the following public NuGet feeds: 
+- [Compiler](https://dev.azure.com/dnceng/public/_packaging?_a=feed&feed=dotnet5): `https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet5/nuget/v3/index.json`
```
Comment (JoeRobich) [inline README.md:27]: Should we mention that to get rolling Compiler builds they would need to reference the dotnet-tools feed instead? dotnet5 seems to track the latest sdk preview.

File: README.md [hunk 2]
```diff
@@ -1,39 +1,9 @@
...
-
-
-[//]: # (End current test results)
+Roslyn provides open-source C# and Visual Basic compilers with rich code analysis APIs.  It enables building code analysis tools with the same APIs that are used by Visual Studio.
```
Comment (333fred) [inline README.md:3]: I feel like badges should go at the top. Most repos I've seen have the discussion links at the top and I don't see any need to break with that form.

Comment (tmat) [reply README.md:3]: Why does anyone care about status of 30 build pipelines?

Comment (tmat) [reply README.md:3]: I don't see any form being broken. https://github.com/dotnet/runtime 

image

Comment (333fred) [reply README.md:3]: I just mean the chat links. And in that example, the gitter link is above the summary.

File: README.md (hunk 1 above)
Comment (tmat) [reply README.md]: Changed the Compiler link to point to dotnet-tools and added .NET SDK as a separate feed.

File: README.md (hunk 2 above)
Comment (tmat) [reply README.md]: Moved chat links higher.

**Pull request merged**